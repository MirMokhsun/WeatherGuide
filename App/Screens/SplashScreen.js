import React, { Component } from 'react'
import { View,Text } from 'react-native'
import {styles} from '../styles/SplashScreenStyles'
import Loader from '../Components/Loader'

export default class SplashScreen extends Component {
    render() {
        return (
            <View style={styles.container}>
                <Loader
                    source={require('../assets/logos/sun.png')} />
                <Text style={styles.txt}>Weather Guide</Text>
            </View>
        )
    }
}

