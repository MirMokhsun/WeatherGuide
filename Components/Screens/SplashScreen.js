import React, { Component } from 'react'
import { Image, View, StyleSheet, Dimensions, Animated, Text } from 'react-native'


const screenWidth = Math.round(Dimensions.get('window').width);
class Loader extends Component {
    state = {
        opacity: new Animated.Value(0),
    }

    onLoad = () => {
        Animated.timing(this.state.opacity, { toValue: 1, duration: 1000, }).start()
    }

    render() {
        return (
            <Animated.Image
                onLoad={this.onLoad}
                {...this.props}
                style={[
                    {
                        width: screenWidth,
                        height: screenWidth,
                        opacity: this.state.opacity,
                        transform: [
                            {
                                scale: this.state.opacity.interpolate({
                                    inputRange: [0, 1],
                                    outputRange: [0.25, 0.75]
                                })
                            }
                        ]
                    }
                ]}
            />
        )
    }
}

export default class SplashScreen extends Component {
    render() {
        return (
            <View style={styles.container}>
                <Loader
                    source={require('../../src/logos/sun.png')} />
                <Text style={styles.txt}>Weather Guide</Text>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#d3f4ff',
    },
    txt: {
        fontSize: 45,
        fontFamily: 'OdibeeSans-Regular'
    }
})