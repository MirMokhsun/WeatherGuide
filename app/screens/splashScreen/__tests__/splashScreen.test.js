import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import SplashScreen from '..';

describe('SplashScreen', () => {
    test('SplashScreen snapshot', () => {
        const renderer = new ShallowRenderer();
        const snap4 = renderer.render(<SplashScreen />);
        expect(snap4).toMatchSnapshot();
    });
});