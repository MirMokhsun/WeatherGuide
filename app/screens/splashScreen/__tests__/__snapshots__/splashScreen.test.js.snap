// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashScreen SplashScreen snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#d3f4ff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Loader
    source={
      Object {
        "testUri": "../../../app/assets/logos/sun.png",
      }
    }
  />
  <Text
    style={
      Object {
        "fontFamily": "OdibeeSans-Regular",
        "fontSize": 45,
      }
    }
  >
    Weather Guide
  </Text>
</View>
`;
